### PR TITLE
Make filepath absolute

### PR DESCRIPTION
**Pull request by mstruebing**
I'm not sure if this is a good addition but I think it is totally okay to pass relative pathnames.

### CODE REVIEW

**Comment by andreynering** [comment]
Hi @mstruebing,

What's the improvement you have in mind with this change? Does this fix any bug or something?

Also, `filepath.Abs` already returns itself if the path is already absolute, so no need to check for that or create a separated function.

**Comment by mstruebing** [comment]
The improvement is that you are not only be able to pass a file with an absolute file path like it is implemented now(https://github.com/editorconfig/editorconfig-core-go/blob/master/cmd/editorconfig/main.go#L45) - with this change you are also able to pass a relative file path and the program continues to run.

File: cmd/editorconfig/main.go [hunk 1]
```diff
@@ -17,10 +17,24 @@ const (
 	Version = "1.2.0"
 )
 
+// Returns the absolute path of a file
+// if it is already absolute it returns
+// the given one
+func getAbsoluteFilePath(path string) string {
```
Comment (andreynering) [inline cmd/editorconfig/main.go]: Let's get rid of this function altogether

File: cmd/editorconfig/main.go [hunk 2]
```diff
@@ -42,11 +56,8 @@ func main() {
 	}
 
 	for _, file := range rest {
-		if !filepath.IsAbs(file) {
-			fmt.Fprintf(os.Stderr, "Input file must be a full path name: %s\n", file)
-			os.Exit(1)
-		}
-		def, err := editorconfig.GetDefinitionForFilenameWithConfigname(file, configName)
+		absolutePath := getAbsoluteFilePath(file)
```
Comment (andreynering) [inline cmd/editorconfig/main.go]:
```go
absolutePath, err := filepath.Abs(file)
if err != nil {
	fmt.Fprintln(os.Stderr, err)
	os.Exit(1)
}
```

File: cmd/editorconfig/main.go (hunk 1 above)
Comment (mstruebing) [reply cmd/editorconfig/main.go]: Thanks, right, of course the function call isn't really needed and the code is also clear without it.

**Pull request merged**